src/agent/agent.c

```diff
@@ -1887,6 +1942,11 @@ static const sd_bus_vtable agent_vtable[] = {
                         agent_property_get_disconnect_timestamp,
                         0,
                         SD_BUS_VTABLE_PROPERTY_EXPLICIT),
+        SD_BUS_PROPERTY("ControllerAddress",
+                        "s",
+                        agent_property_get_controller_address,
+                        0,
+                        SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```

Comment:
```suggestion
        SD_BUS_PROPERTY("ControllerAddress",
                        "s",
                        NULL,
                        offsetof(Agent, orch_addr),
                        SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```

Instead of implementing a dedicated function we can use `offsetof` since the property is a field on the `Agent` struct. So `agent_property_get_controller_address` can be removed. 

Also, I'd use `agent->orch_addr` instead of `agent->controller_address` for the property. The "final" address of the controller where the agent connects to is assembled either from
- `host` and `port`
- or `controller_address`

and saved to `agent->orch_addr`. So by using this field, the property shows the actual address used.

Reply:
Thanks for the feedback, i updated.

Reply:
Thanks, looks good!